scripts/test/Muon/elemental_analysis/name_generator_test.py

```diff
@@ -0,0 +1,98 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+from __future__ import absolute_import, print_function
+
+import unittest
+import sys
+
+from Muon.GUI.ElementalAnalysis.elemental_analysis import gen_name
+
+
+class NameGeneratorTest(unittest.TestCase):
+    def test_that_gen_name_returns_name_if_it_contains_element(self):
+        element_name = 'name2'
+        name = 'name1name2name3nam4'
+
+        self.assertEqual(gen_name(element_name, name), 'name1name2name3nam4')
+
+    def test_that_gen_name_combines_element_and_label(self):
+        element = 'element'
+        name = 'label'
+
+        self.assertEqual(gen_name(element, name), 'element label')
+
+    def test_that_gen_name_with_non_string_element_throws(self):
+        element1 = None
+        element2 = 1
+        element3 = (3.0, 'string')
+        name = 'valid_name'
+
+        with self.assertRaises(TypeError) as err:
+            gen_name(element1, name)
+        if sys.version_info[:2] < (3, 0):
```

Comment:
we could use a function here to make it easier to read (I think the below should work):
```
def exception_produced(self, err, element, name, error_str):
with self.assertRaises(TypeError) as err:
            gen_name(element, name)
        if sys.version_info[:2] < (3, 0):
            self.assertEqual(str(err.exception),
                             error_str+" expected element to be 'str', found '<type 'int'>' instead")
        else:
            self.assertEqual(str(err.exception),
                             error_str+" expected element to be 'str', found '<class '" +type(name) +"'>' instead")
```